pkg/dev/logs.go

```diff
@@ -0,0 +1,98 @@
+package dev
+
+import (
+	"sync"
+
+	"github.com/airplanedev/cli/pkg/api"
+)
+
+// LogWatcher is a receiver for logs. There can be many log watchers for a given run.
+type LogWatcher struct {
+	LogChannel  chan api.LogItem
+	DoneChannel chan struct{}
+	// replayOnce is used to ensure we replay past logs only once.
+	replayOnce *sync.Once
+}
+
+// LogBroker keeps track of a set of log watchers and streams logs to them. There should only be one log broker per run.
+type LogBroker struct {
+	// Watchers is a set of all log watchers.
+	Watchers map[LogWatcher]bool
```

Comment:
Clarifies that the map value has no meaning.

```suggestion
	Watchers map[LogWatcher]struct{}
```